pkgs/build-support/rust/rust-utils.nix

```diff
@@ -0,0 +1,270 @@
+# Copyright 2017 Pierre-Étienne Meunier
+#
+# This file is licensed under the Apache-2.0 license, and under the
+# MIT license, at your convenience.
+
+{ lib, buildPlatform, stdenv }:
+
+let buildCrate = { crateName, crateVersion, dependencies, complete, crateFeatures, libName, build, release, libPath, crateType, metadata, crateBin, finalBins, verboseBuild }:
+
+      let depsDir = builtins.foldl' (deps: dep: deps + " " + dep.out) "" dependencies;
+          completeDepsDir = builtins.foldl' (deps: dep: deps + " " + dep.out) "" complete;
+          deps =
+            builtins.foldl' (deps: dep:
+              let extern = lib.strings.replaceStrings ["-"] ["_"] dep.libName; in
+              deps + (if dep.crateType == "lib" then
+                 " --extern ${extern}=${dep.out}/lib${extern}-${dep.metadata}.rlib"
+              else
+                 " --extern ${extern}=${dep.out}/lib${extern}-${dep.metadata}.so")
+            ) "" dependencies;
+          optLevel = if release then 3 else 0;
+          rustcOpts = (if release then "-C opt-level=3" else "-C debuginfo=2");
+          rustcMeta = "-C metadata=" + metadata + " -C extra-filename=-" + metadata;
+      in ''
+
+      norm="$(printf '\033[0m')" #returns to "normal"
+      bold="$(printf '\033[0;1m')" #set bold
+      green="$(printf '\033[0;32m')" #set red
+      boldgreen="$(printf '\033[0;1;32m')" #set bold, and set red.
```

Comment:
Better use tput here. Otherwise it might print garbage if stdout is not a terminal.

Reply:
I'm trying to do this, but I keep getting the error

```
tput: No value for $TERM and no -T specified
```

When I set a TERM to my terminal, I get instead:

```
tput: terminal attributes: No such device or address
```

The command I'm trying to use is
```
         echo "`tput setaf 2``tput bold`Building $BUILD (${libName})`tput sgr0`"
```